meraki_appliance.tf

```
@@ -444,7 +444,7 @@ locals {
             network_id      = meraki_network.organizations_networks[format("%s/%s/%s", domain.name, organization.name, network.name)].id
             vlan_id         = try(appliance_vlan.vlan_id, local.defaults.meraki.domains.organizations.networks.appliance.vlans.vlan_id, null)
             appliance_ip    = try(appliance_vlan.appliance_ip, local.defaults.meraki.domains.organizations.networks.appliance.vlans.appliance_ip, null)
-            group_policy_id = try(appliance_vlan.group_policy_id, local.defaults.meraki.domains.organizations.networks.appliance.vlans.group_policy_id, null)
+            group_policy_id = try(meraki_network_group_policy.networks_group_policies[format("%s/%s/%s/%s", domain.name, organization.name, network.name, try(appliance_vlan.group_policy_name, local.defaults.meraki.domains.organizations.networks.appliance.vlans.group_policy_name))].id, null)
```

We don't apply defaults for names and name-to-id mappings.
```suggestion
            group_policy_id = try(meraki_network_group_policy.networks_group_policies[format("%s/%s/%s/%s", domain.name, organization.name, network.name, appliance_vlan.group_policy_name)].id, null)
```